File: bikewear_app/backend/Models/WearPart.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace App.Models
{
    public enum WearPartCategory
    {
        Reifen,
        Kassette,
        Kettenblatt,
        Kette,
        Sonstiges,
        Federung
    }

    public class WearPart
    {
        [Key]
        public int Id { get; set; }
        [Required]
        public int RadId { get; set; }
        [Required]
        public string Name { get; set; } = string.Empty;
        public WearPartCategory Kategorie { get; set; }
        public string? Position { get; set; }
        public int EinbauKilometerstand { get; set; }
        public int? AusbauKilometerstand { get; set; }
        public DateTime EinbauDatum { get; set; }
        public DateTime? AusbauDatum { get; set; }

        /// <summary>Fahrstunden des Rades zum Einbauzeitpunkt (nur für Federung relevant).</summary>
        public double? EinbauFahrstunden { get; set; }

        /// <summary>Fahrstunden des Rades zum Ausbauzeitpunkt (nur für Federung relevant).</summary>
        public double? AusbauFahrstunden { get; set; }

        public string? Notizen { get; set; }
    }
}

File: bikewear_app/frontend/src/app/models/wear-part.ts
import { WearPartCategory } from './wear-part-category';

export interface WearPart {
  id: number;
  radId: number;
  name: string;
  kategorie: WearPartCategory;
  position: string | null;
  einbauKilometerstand: number;
  ausbauKilometerstand: number | null;
  einbauDatum: Date;
  ausbauDatum: Date | null;
  einbauFahrstunden: number | null;
  ausbauFahrstunden: number | null;
  notizen: string | null;
  vorgaengerId: number | null;
  gruppeId: number | null;
  reifenBreiteMm?: number | null;
  reifenBreiteZoll?: number | null;
  reifenDruckBar?: number | null;
  reifenDruckPsi?: number | null;
}

export interface MoveWearPartRequest {
  zielRadId: number;
  ausbauKilometerstand: number;
  ausbauDatum: Date;
  einbauKilometerstand: number;
  einbauDatum: Date;
  ausbauFahrstunden?: number | null;
  einbauFahrstunden?: number | null;
}